frontend/src/lib/utils.tsx

```diff
@@ -1677,3 +1677,14 @@ export function shouldCancelQuery(error: any): boolean {
     // the query will continue running in ClickHouse
     return error.name === 'AbortError' || error.message?.name === 'AbortError' || error.status === 504
 }
+
+export function debounce<F extends (...args: Parameters<F>) => ReturnType<F>>(
```

Comment:
Would the [`use-debounce` package](https://github.com/xnimorz/use-debounce#debounced-callbacks) we already have installed handle this?

Reply:
Weird. I searched and didn't find it. Looks like exactly what I need though 👍